File: triagem-app/src/app/domain/repository/processo.repository.ts
import { Observable } from 'rxjs';
import { Processo } from '../models/processo.model';
import { StatusProcesso } from '../models/status-processo.enum';
import { Pagina } from '../models/pagina.model';

export interface ProcessoRepository {
  listarPaginado(pagina: number, tamanho: number): Observable<Pagina<Processo>>;
  salvar(processo: Partial<Processo>): Observable<Processo>;
  atualizarStatus(id: number, status: StatusProcesso): Observable<Processo>;
}

File: triagem-api/src/main/java/com/attus/triagem_api/domain/repository/ProcessoRepository.java
package com.attus.triagem_api.domain.repository;

import com.attus.triagem_api.domain.model.Pagina;
import com.attus.triagem_api.domain.model.Processo;

import java.util.List;
import java.util.Optional;

public interface ProcessoRepository {
    Processo salvar(Processo processo);
    Pagina<Processo> listarPaginado(Integer pagina, Integer tamanho);
    Optional<Processo> buscarPorId(Long id);
    Optional<Processo> buscarPorNumero(String numeroProcesso);
}